exporter/awsemfexporter/metric_translator.go

```diff
@@ -118,6 +134,268 @@ func (dps DoubleHistogramDataPointSlice) At(i int) DataPoint {
+							zap.String("DataType", metric.DataType().String()),
+							zap.String("Name", metric.Name()),
+							zap.String("Unit", metric.Unit()),)
+						continue
+				}
+
+				if dps.Len() == 0 {
+					continue
+				}
+
+				for m := 0; m < dps.Len(); m++ {
+					dp := dps.At(m)
+					if dp.IsNil() {
+						continue
+					}
+
+					fields := make(map[string]interface{})
+					fields[nameSpace] = namespace
+					fields[OTellibDimensionKey] = instrumentationLibName
+	
+					dp.LabelsMap().ForEach(func(k string, v string) {
+						fields[k] = v
+					})
+					
+					s := make([]string, len(fields)*2)
+					for k, v := range fields {
+						if (k != OTellibDimensionKey) {
+							s = append(s, k, v.(string))
+						} else {
+							s = append(s, k, v.(string))
+						}
+					}
+					sort.Strings(s)
+					key := strings.Join(s, "")
+
+					if _, ok := groupedMetricMap[key]; ok {
+						updateGroupedMetric(dp, &metric, key, groupedMetricMap)
+					} else {
+						groupedMetric := buildGroupedMetric(dp, fields, &metric, config.DimensionRollupOption)
+						if groupedMetric != nil {
+							groupedMetricMap[key] = groupedMetric
+						}
+					}
+				}
+			}
+		}
+	}
+	return groupedMetricMap, totalDroppedMetrics
+}
+
+// Build grouped metric from Datapoint and pdata.Metric
+func buildGroupedMetric (dp DataPoint, fields map[string]interface{}, pMetricData *pdata.Metric, dimensionRollupOption string) (*GroupedMetric) {
+	var namespace string
+	timestamp := time.Now().UnixNano() / int64(time.Millisecond)
+	labels := make(map[string]string)
+	metrics := make(map[string]MetricInfo)
+	
+	namespace = fields[nameSpace].(string)
+
+	// Extract labels
+	for k, v := range fields {
+		if k != nameSpace {
+			labels[k] = v.(string)
+		}
+	}
+
+	// Extract metric
+	var metricVal interface{}
+	switch metric := dp.(type) {
+	case pdata.IntDataPoint:
+		metricVal = int64(metric.Value())
+		if needsCalculateRate(pMetricData) {
+			metricVal = calculateRate(fields, metric.Value(), timestamp)
+		}
+	case pdata.DoubleDataPoint:
+		metricVal = float64(metric.Value())
+		if needsCalculateRate(pMetricData) {
+			metricVal = calculateRate(fields, metric.Value(), timestamp)
+		}
+	case pdata.DoubleHistogramDataPoint:
+		bucketBounds := metric.ExplicitBounds()
+		metricVal = &CWMetricStats{
+			Min:   bucketBounds[0],
+			Max:   bucketBounds[len(bucketBounds)-1],
+			Count: metric.Count(),
+			Sum:   metric.Sum(),
+		}
+	}
+	if metricVal == nil {
+		return nil
+	}
+	metricInfo := &MetricInfo {
+		Value: metricVal,
+		Unit:  pMetricData.Unit(),
+	}
+
+	metrics[pMetricData.Name()] = *metricInfo
+
+	groupedMetric := &GroupedMetric {
+		Namespace: namespace,
+		Timestamp: timestamp,
+		Labels:    labels,
+		Metrics:   metrics,
+	}
+	return groupedMetric
+}
+
+// Update GroupedMetric with new metric from datapoint
+func updateGroupedMetric (dp DataPoint, pMetricData *pdata.Metric, key string, groupedMetricMap map[string]*GroupedMetric) {
```

Comment:
I'm unclear about this function's purpose. Why are we overriding past metrics? Should we instead be appending on to a list of metrics?